crates/ide-assists/src/handlers/auto_import.rs

```diff
@@ -158,11 +165,142 @@ fn group_label(import_candidate: &ImportCandidate) -> GroupLabel {
     GroupLabel(name)
 }
 
+/// Determine how relevant a given import is in the current context. Higher scores are more
+/// relevant.
+fn relevance_score(ctx: &AssistContext, import: &LocatedImport) -> i32 {
+    let mut score = 0;
+
+    let db = ctx.db();
+
+    let item_module = match import.item_to_import {
+        hir::ItemInNs::Types(item) | hir::ItemInNs::Values(item) => item.module(db),
+        hir::ItemInNs::Macros(makro) => Some(makro.module(db)),
+    };
+
+    let current_node = match ctx.covering_element() {
+        NodeOrToken::Node(node) => Some(node),
+        NodeOrToken::Token(token) => token.parent(),
+    };
+
+    if let Some(module) = item_module.as_ref() {
+        if module.krate().is_builtin(db) {
+            // prefer items builtin to the language
+            score += 5;
+        }
+    }
+
+    match item_module.zip(current_node) {
+        // get the distance between the modules (prefer items that are more local)
+        Some((item_module, current_node)) => {
+            let current_module = ctx.sema.scope(&current_node).unwrap().module();
```

Comment:
Drive-by nit: I don't know if `ctx.sema.scope(&current_node)` can be missing here, but we've had `unwrap`s fail before and here it seems possible to avoid it.

Reply:
Right, I was too focused to get things working I forgot to fix every unwrap.